lib/index.js

```diff
@@ -207,10 +224,20 @@ api.set = async ({
   } else {
     query['account.email'] = email;
   }
-  const result = await database.collections.account.updateMany(
+
+  let tokenTypeValue;
+  if(type === 'nonce') {
+    const oldValue = tokens.length > 0 ? tokens.slice() : {$in: [null, []]};
+    query['meta.bedrock-authn-token.tokens.nonce'] = oldValue;
+    tokens.push(token);
+    tokenTypeValue = tokens;
+  } else {
+    tokenTypeValue = token;
+  }
+  const result = await database.collections.account.updateOne(
     query, {
       $set: {
-        [`meta.bedrock-authn-token.tokens.${type}`]: token
+        [`meta.bedrock-authn-token.tokens.${type}`]: tokenTypeValue
       }
     }, database.writeOptions);
   if(result.result.n === 0) {
```

Comment:
Within this conditional, if the type is nonce, we should return a different error:

```js
  if(result.result.n === 0) {
    if(type === 'nonce') {
      throw new BedrockError(
        'Could not set account "nonce" token; another "nonce" token ' +
        'was set concurrently.',
        'DuplicateError', {httpStatusCode: 409, public: true});
    }
    throw new BedrockError(
      'Could not set account token. Account not found.',
      'NotFoundError', {httpStatusCode: 404, public: true});
  }
```